web/src/js/index.js

```diff
@@ -1,473 +1,479 @@
+
+      // auto select if only 1 option exists
+      if (this.resourceList.length === 1) {
+        this.selectedResource = this.resourceList[0];
+        this.updateResourceId(service, this.resourceList[0]);
+      } else {
+        this.resourceIdPlaceholder = "";
+      }
+    },
+    openAllowlistDeletePopup: function (resourceId) {
+      this.selectedResourceId = resourceId;
+      this.showAllowlistDeletePopup = true;
+      this.resourceIdPlaceholder = "";
+    },
+    openAllowlistInsertPopup: function () {
+      this.showAllowlistPopup = true;
+      this.resourceIdPlaceholder = "";
+    },
+    searchAllowlist: function () {
+      this.allowlistDataTables.search(this.allowlistSearchTerm).draw();
+    },
+    // Execution Log
+    closeExecutionLogPopup: function () {
+      $("html").removeClass("remove-overflow");
+
+      this.executionLogDataTables.destroy();
+
+      this.executionLogActionStats = {};
+      this.executionLogRegionStats = {};
+      this.executionLogServiceStats = {};
+      this.showExecutionLogPopup = false;
+    },
+    openExecutionLog: function (keyURL) {
+      $("html").addClass("remove-overflow");
+
+      getExecutionLog(keyURL);
+    },
+    searchExecutionLog: function () {
+      this.executionLogDataTables.search(this.executionLogSearchTerm).draw();
+    },
+    // Help
+    closeHelpPopup: function () {
+      this.showHelpPopup = false;
+    },
+    openHelpPopup: function () {
+      this.showHelpPopup = true;
+    },
+    // Api Key
+    closeApiKeyPopup: function () {
+      this.showApiKeyPopup = false;
+    },
+    setApiKey: function () {
+      this.showApiKeyPopup = false;
+      API_KEY = this.apiKey;
+      localStorage.setItem("x-api-key", this.apiKey);
+      init();
+    },
+    resetApiKey: function () {
+      API_KEY = "";
+      localStorage.removeItem("x-api-key");
+      location.reload();
+    },
+  },
+  mounted: function () {
+    API_KEY = localStorage.getItem("x-api-key");
+    if (API_KEY !== null) {
+      this.showApiKeyPopup = false;
+    }
+  },
+});
+
+function sendApiRequest(formURL, requestMethod) {
+  fetch(API_CRUD_ALLOWLIST + "?" + formURL, {
+    method: requestMethod,
+    headers: {
+      "x-api-key": API_KEY,
+    },
+  })
+    .then((response) => response.json())
+    .then((data) => {
+      refreshAllowlist();
+      app.closeAllowlistInsertPopup();
+      app.closeAllowlistDeletePopup();
+
+      iziToast.success({
+        color: "#3FBF61",
+        message: data.message,
+        messageColor: "white",
+      });
+    })
+    .catch((error) => {
+      iziToast.error({
+        color: "#EC2B55",
+        message: error,
+        messageColor: "white",
+        title: "Something went wrong",
+      });
+    });
+}
+
+// Get execution log for a single instance
+function getExecutionLog(executionLogUrl) {
+  app.showExecutionLogPopup = true;
+  app.showExecutionLogLoadingGif = true;
+
+  fetch(API_EXECLOG + executionLogUrl, {
```

Comment:
Codacy has a fix for the issue: [Delete `␍`](https://app.codacy.com/gh/servian/aws-auto-cleanup/pullRequest?prid=9792602)
```suggestion
  fetch(API_EXECLOG + executionLogUrl, {
```